std/stdio.d

```diff
@@ -651,15 +652,15 @@ Detaches from the underlying file. If the sole owner, calls $(D close).
 
 Throws: $(D ErrnoException) on failure if closing the file.
   */
-    void detach() @safe
+    void detach() @trusted
     {
         if (!_p) return;
-        if (_p.refs == 1)
+        if (atomicLoad(_p.refs) == 1)
             close();
         else
```

Comment:
> The correct way to do this is to atomically decrement, and check for 0.

I don't think that's possible, as close expects the refcount to be 1, and close can be called by the programmer. 

The only option I see now is locking in detach.